src/pivot-table/hooks/use-column-width.ts

```diff
@@ -97,59 +100,81 @@ export default function useColumnWidth(
-            return getColumnWidth(columnWidth, fitToContentWidth);
-          }),
-        );
-      } else {
-        const { label, qApprMaxGlyphCount, columnWidth, isLocked } = cell;
-        const expandIconSize = !isFullyExpanded && index < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
-        const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
-
-        const fitToContentWidth =
-          TOTAL_CELL_PADDING +
-          Math.max(
-            measureTextForHeader(label) + MENU_ICON_SIZE + lockedIconSize,
-            estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + expandIconSize,
+      let sumOfWidths = 0;
+
+      const lastRow = headersData.data[headersData.size.y - 1] as HeaderCell[];
+      const columnWidths = lastRow.map((cell, index) => {
+        let width: number;
+
+        if (widthOverride && overrideIndex !== undefined && overrideIndex === index) {
+          width = widthOverride;
+        } else if (cell.id === PSEUDO_DIMENSION_KEY) {
+          // Use the max width of all measures
+          width = Math.max(
+            ...qMeasureInfo.map(({ qFallbackTitle, columnWidth }) => {
+              const fitToContentWidth = measureTextForDimensionValue(qFallbackTitle) + TOTAL_CELL_PADDING;
+              return getColumnWidth(columnWidth, fitToContentWidth);
+            }),
           );
+        } else {
+          const { label, qApprMaxGlyphCount, columnWidth, isLocked } = cell;
+          const expandIconSize = !isFullyExpanded && index < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
+          const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
+
+          const fitToContentWidth =
+            TOTAL_CELL_PADDING +
+            Math.max(
+              measureTextForHeader(label) + MENU_ICON_SIZE + lockedIconSize,
+              estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + expandIconSize,
+            );
+
+          width = getColumnWidth(columnWidth, fitToContentWidth);
+        }
 
-        width = getColumnWidth(columnWidth, fitToContentWidth);
-      }
+        sumOfWidths += width;
+        return width;
+      });
 
-      sumOfWidths += width;
-      return width;
-    });
+      return {
+        leftGridWidth: Math.min(rect.width * LEFT_GRID_MAX_WIDTH_RATIO, sumOfWidths),
+        leftGridColumnWidths: columnWidths,
+        leftGridFullWidth: sumOfWidths,
+      };
+    },
+    [
+      estimateWidthForDimensionValue,
+      headersData.data,
+      headersData.size.y,
+      isFullyExpanded,
+      measureTextForDimensionValue,
+      measureTextForHeader,
+      qMeasureInfo,
+      qNoOfLeftDims,
+      rect.width,
+    ],
+  );
 
-    return {
-      leftGridWidth: Math.min(rect.width * LEFT_GRID_MAX_WIDTH_RATIO, sumOfWidths),
-      leftGridColumnWidths: columnWidths,
-      leftGridFullWidth: sumOfWidths,
-    };
+  const [leftGridWidthInfo, setLeftGridWidthInfo] = useState<LeftGridWidthInfo>(calculateLeftGridWidthInfo());
```

Comment:
Did you end up calling the function anyway?

Reply:
yes, to avoid the confusion

Reply:
See my comment about the performance impact of this

Reply:
fixed